src/reducers/example.tsx

```diff
@@ -5,7 +5,7 @@ type TogglePayload = {};
 
 export const actions = {
   toggle: createAction('TOGGLE', (resolve) => {
-    return () => resolve({} as TogglePayload);
```

Comment:
I see that this fails a rule but we need to provide type hints here (it's the way `typesafe-actions` wants us to do it). Maybe there's another way?

Here is a hypothetical problem that we would need the typehint to catch for us (the resolved action payload is invalid):

```diff
diff --git a/src/reducers/example.tsx b/src/reducers/example.tsx
index d2c6c48..2667eb2 100644
--- a/src/reducers/example.tsx
+++ b/src/reducers/example.tsx
@@ -1,19 +1,21 @@
 import { Reducer } from 'redux';
 import { ActionType, createAction, getType } from 'typesafe-actions';
 
-type TogglePayload = {};
+type TogglePayload = { thing: 'string' };
 
 export const actions = {
   toggle: createAction('TOGGLE', (resolve) => {
-    return () => resolve({});
+    return () => resolve({ thing: 'something', aTypo: 'should-fail' });
   }),
 };
 
 export type ExampleState = {
+  thing: string,
   toggledOn: boolean;
 };
 
 const initialState: ExampleState = {
+  thing: '',
   toggledOn: false,
 };
 
@@ -23,7 +25,7 @@ const reducer: Reducer<ExampleState, ActionType<typeof actions>> = (
 ): ExampleState => {
   switch (action.type) {
     case getType(actions.toggle):
-      return { ...state, toggledOn: !state.toggledOn };
+      return { ...state, toggledOn: !state.toggledOn, thing: action.payload.thing };
     default:
       return state;
   }
```


Using a type hint will catch the error:

```diff
diff --git a/src/reducers/example.tsx b/src/reducers/example.tsx
index d2c6c48..7cc8cca 100644
--- a/src/reducers/example.tsx
+++ b/src/reducers/example.tsx
@@ -1,19 +1,21 @@
 import { Reducer } from 'redux';
 import { ActionType, createAction, getType } from 'typesafe-actions';
 
-type TogglePayload = {};
+type TogglePayload = { thing: 'string' };
 
 export const actions = {
   toggle: createAction('TOGGLE', (resolve) => {
-    return () => resolve({});
+    return () => resolve({ thing: 'something', aTypo: 'should-fail' } as TogglePayload);
   }),
 };
 
 export type ExampleState = {
+  thing: string,
   toggledOn: boolean;
 };
 
 const initialState: ExampleState = {
+  thing: '',
   toggledOn: false,
 };
 
@@ -23,7 +25,7 @@ const reducer: Reducer<ExampleState, ActionType<typeof actions>> = (
 ): ExampleState => {
   switch (action.type) {
     case getType(actions.toggle):
-      return { ...state, toggledOn: !state.toggledOn };
+      return { ...state, toggledOn: !state.toggledOn, thing: action.payload.thing };
     default:
       return state;
   }
```

```
/Users/kumar/dev/addons-code-manager/src/reducers/example.tsx
Type error: Conversion of type '{ thing: "something"; aTypo: string; }' to type 'TogglePayload' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Types of property 'thing' are incompatible.
    Type '"something"' is not comparable to type '"string"'.  TS2352

     6 | export const actions = {
     7 |   toggle: createAction('TOGGLE', (resolve) => {
  >  8 |     return () => resolve({ thing: 'something', aTypo: 'should-fail' } as TogglePayload);
       |                          ^
     9 |   }),
    10 | };
    11 |
```

Reply:
right, but this is useless here, no? There is nothing in the payload? I don't even think we need to add the `resolve` function.

Reply:
That was useful to implement the `api` action though, so thanks!

Reply:
Yes, it's useless in the real code but I was just pointing out how we will need to use this pattern (or some alternative) in the future. It was a little different in your auth patch since payload was the input parameter and could be typed that way.